### ISSUE TITLE
Facility Locator: Update " No results" content and behavior

### ISSUE THREAD

**Issue by laflannery** [department-of-veterans-affairs/va.gov-cms#18439]
Eli to run a11y testing. Thus: Randi / Bryan assigned and whoever gets to it first...wins.

## Background
In an effort to move the facility locator redesign work forward, we are trying to break up the large project into smaller, more manageable pieces that can be completed more easily and added into sprints.

## Description
Currently, when there are no results found in the facility locator a message is shown, however the behavior and content of this message could be more helpful.

[Link to Figma file](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=175-31801&m=dev)

**Current message**
This is used on Desktop and Mobile listing. Mobile Map doesn't have a no results message.
![image](https://github.com/department-of-veterans-affairs/va.gov-cms/assets/106678594/c554a1b7-8b05-4962-86ad-055c2b66945c)

### New message
There are 2 versions of the new message. See the [link to Figma file](https://www.figma.com/design/GvNVeG9GAXZaBpny7vNxZP/Facility-Locator?node-id=175-31801&m=dev) for clarification on where these messages should be displayed

No results message 1 will be used for Desktop and Mobile map

> Try searching for something else or in a different area.
> **Suggestions:**
> - Zoom out to view larger area of the map
> - Move the map to a different area
> - Enter a different location, facility type, or service type

No results message 2 will be used for for the mobile listing

> Search for something else or in a different area. Try entering a different location, facility type, or service type

### Functionality notes
- Currently the no results message shows before the user has even filled out all their filters, which is odd and confusing behavior.
  - For example, I put in my location, then I select "VA health" as my facility and then the No results message shows. This doesn't make sense because I haven't tried to search yet, why are you telling me how to get better results? 
  - We should update this behavior so that this message **only** shows when there are no results and otherwise, if the user has not yet hit the search button, the default text still displays (Please enter a location (street, city, state, or postal code) and facility type, then click search above to find facilities.).
- Also, recently Eli and I worked to update the tabindex of the no results messages and we need to be sure there are no regressions here. The way it currently works in Prod:
  - When a user completes a search, there is a message at the top of the listing that gets focus and says "Showing 1 - 10 of XX results for {all the filter variables}". Or if there are no results the message says "No results found for {all the filter variables}"
  - Focus is forced to this message once the search is complete
  - The next tab stop is:
    - If there are search results, the first interactive element in the results (Facility name, Get Directions link, etc.)
    - If there are no results, the entire section that contains the no results message
  - The 2 cases where this behavior differs is Emergency Care and Urgent care. [This is documented in a separate ticket](https://github.com/department-of-veterans-affairs/va.gov-cms/issues/19176) so those are the only instances where the tabindex/focus behavior should differ from the above pattern.

## Acceptance Criteria
- [x] No results content has been updated for all scenarios to match Figma file
- [x] All text is wrapped in semantic tags so that the font resizes and scales appropriately
- [x] No results message does not display unless the user has activated the search button
- [x] There are no regressions in the tabindex nor focus management for no results message or the results listings for any of the Facility types
- [x] Requires design review
- [x] Requires accessibility review

## Accessibility review steps
**Confirm the font in all no results messages resize with the browser font settings**
1. Go to the Facility Locator page `/find-locations`
2. Complete a search, so that the results list appears
3. Within your default browser, update your settings to the largest font size available (you do not need to test all of these):
      - Chrome: Settings > Appearance > Font size
      - Firefox: Settings > Language & Appearance > Fonts
      - Edge: Settings > Appearance > Fonts
4. Confirm that for each item in the description, the text size increases correctly

**Confirm the no results message is read properly by VoiceOver**
(because I believe the majority of people on the team have Macs)
1. Go to the Facility Locator page `/find-locations`
2. Turn on VO: CMD + FN + F5
3. Using the keyboard, complete a search, such that no results appear
4. When the search completes and the no results message appears, use the tab key to tab into this area and confirm VO begins reading this message.

**Confirm the tab order is correct**
1. Go to the Facility Locator page `/find-locations`
2. Complete the following steps for each Facility type
   1. Complete a search, so that the results list appears
   2. Confirm that focus is brought directly to the "Showing X - X..." message
      - Except Emergency Care and Urgent Care, focus will remain on the Search button 
   3. Confirm that the next tab stop is the next interactive element directly after this message
   4. Now complete a search, so that the no results message appears
   5. Confirm that focus is brought directly to the "No results found for {variables}" message
   6. Confirm that the next tab stop is the new no results suggestions message

**Comment by Agile6MSkinner** [comment]
@thejordanwood to add clarification around mobile vs. desktop "No results" message.

**Comment by thejordanwood** [comment]
@Agile6MSkinner Laura got to this first and added the new mobile "No results" messaging we want to use to the ticket body. 

- The mobile list view will need the new messaging
- The mobile map view currently doesn't show any "No results" message, but adding that would be a good improvement. 

cc: @laflannery

**Comment by laflannery** [comment]
@Agile6MSkinner Just throwing out that I do have mild anxiety because of the focus management piece of this and making sure there are no regressions there. The most important thing is that nothing should changed from how this works in Prod now.

My recommendation would probably be that because @eselkin has the most knowledge on this, would they would feel comfortable leading accessibility review in my absence?

**Comment by jilladams** [comment]
CAIA has accessibility experts available who can help us with pointers on testing if we run into trouble. THey live in #accessibility-help

**Comment by randimays** [comment]
Verified successfully in production; closing as complete.
